File: exercises/python_playwright/Class_Attribute/class_attribute_playwright.py
from playwright.sync_api import Dialog, Page


class ClassAttribute:
    def __init__(self, page: Page):
        """Initialize the page object with Playwright locators."""
        self.page = page
        self.button = page.locator("button.btn-primary")

    def open(self) -> None:
        """Navigate to the Class Attribute exercise page."""
        self.page.goto("http://uitestingplayground.com/classattr")

    def click_primary_button(self) -> str:
        """Click the primary button and return the following alert message."""
        dialog_msg = ""

        def dialog_handler(dialog: Dialog) -> None:
            nonlocal dialog_msg
            dialog_msg = dialog.message
            dialog.accept()

        self.page.once("dialog", dialog_handler)
        self.button.click()

        return dialog_msg

File: exercises/javascript_playwright/class-attribute/classAttributePlaywright.js
export class ClassAttrPage {
    /**
     *
     * @param {import('@playwright/test').Page} page
     */
    constructor(page) {
        this.page = page;
        this.button = page.locator(".btn-primary");
    }

    async openExercisePage() {
        await this.page.goto("http://uitestingplayground.com/classattr");
    }

    async clickPrimaryButtonAndAcceptAlert() {
        let dialogMessage = "";

        this.page.once("dialog", async (dialog) => {
            dialogMessage = dialog.message();
            await dialog.accept();
        });

        await this.button.click();

        return dialogMessage;
    }
}
